Manual updates 20221025 material3

moljac opened the pull request:
### Does this change any of the generated binding API's?

Yes. (New packages and dependencies)

### Describe your contribution

Bound `material3` and dependencies:

- `androidx.compose.animation:animation` - 1.2.1 -> 1.3.0
- `androidx.compose.animation:animation-core` - 1.2.1 -> 1.3.0
- `androidx.compose.animation:animation-graphics` -  -> 1.3.0
- `androidx.compose.foundation:foundation` - 1.2.1 -> 1.3.0
- `androidx.compose.foundation:foundation-layout` - 1.2.1 -> 1.3.0
- `androidx.compose.material:material` - 1.2.1 -> 1.3.0
- `androidx.compose.material:material-icons-core` - 1.2.1 -> 1.3.0
- `androidx.compose.material:material-icons-extended` - 1.2.1 -> 1.3.0
- `androidx.compose.material:material-ripple` - 1.2.1 -> 1.3.0
- `androidx.compose.material3:material3` -  -> 1.0.0
- `androidx.compose.material3:material3-windowsizeclass` -  -> 1.0.0
- `androidx.compose.runtime:runtime` - 1.2.1 -> 1.3.0
- `androidx.compose.runtime:runtime-livedata` - 1.2.1 -> 1.3.0
- `androidx.compose.runtime:runtime-rxjava2` - 1.2.1 -> 1.3.0
- `androidx.compose.runtime:runtime-rxjava3` -  -> 1.3.0
- `androidx.compose.runtime:runtime-saveable` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-geometry` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-graphics` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-text` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-tooling` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-tooling-data` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-tooling-preview` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-unit` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-util` - 1.2.1 -> 1.3.0
- `androidx.compose.ui:ui-viewbinding` - 1.2.1 -> 1.3.0
- `com.google.android.material:compose-themeadapter-3` -  -> 1.0.18

moljac commented:
https://github.com/xamarin/AndroidX/issues/637

jpobst reviewed (commented):
For all the other "compose" packages we used this to prevent managed bindings from being created:

```xml
<remove-node path="/api/package" />
```

Should we be doing that with these compose packages as well?

moljac commented:
> For all the other "compose" packages we used this to prevent managed bindings from being created:
> 
> ```
> <remove-node path="/api/package" />
> ```
> 
> Should we be doing that with these compose packages as well?

I am a little bit confused with interrogative form of the comment. Namely, I am aware that Compose cannot be used from managed code (for now, if ever), but I don't remember adding (and the reason for adding):

```xml
<remove-node path="/api/package" />
```

Checking git blame I see you added it for namespaces

https://github.com/xamarin/AndroidX/blame/4977973e4c935dfbcfe2e5f0b9e6853e59767342/source/androidx.compose.animation/animation-core/Transforms/Metadata.xml#L2

Use <AndroidNamespaceReplacement> instead of metadata to rename namespaces

[Use <AndroidNamespaceReplacement> instead of metadata to rename nam](https://github.com/xamarin/AndroidX/commit/0aaa542a8fdc05649d76c8dacc0be51e71a33784)

jpobst commented:
> `<remove-node path="/api/package" />`

I think it was added in the PR where we added those packages:
https://github.com/xamarin/AndroidX/pull/352

I simply moved them from `Metadata.Namespaces.xml` to `Metadata.xml`:
https://github.com/xamarin/AndroidX/commit/0aaa542a8fdc05649d76c8dacc0be51e71a33784#diff-9cb222843becbdac30f925bae0bd249c2db654bace8140128451a757b47ed24e

moljac commented:
> > `<remove-node path="/api/package" />`
> 
> I think it was added in the PR where we added those packages: #352

Could be. I don't remember. I changed according to suggestions.

jpobst reviewed (approved):
LGTM, please add a comment on the PR for why `com.google.android.material:compose-theme-adapter-3` is marked as `frozen` so we have it documented somewhere.

MagicAndre1981 commented:
@moljac @jpobst the packages from this PR miss the new description from #623 

![image](https://user-images.githubusercontent.com/8274816/199333772-4090b77b-3ec2-42f6-a66d-a09c4a401dcd.png)

What went wrong here?
